coordinator/cluster/cluster.go

```diff
@@ -134,16 +145,12 @@ func (c *LoadAgentCluster) IncrementUsers(n int) error {
 	}
 	for i, inc := range dist {
 		c.log.Info("cluster: adding users to agent", mlog.Int("num_users", inc), mlog.String("agent_id", c.config.Agents[i].Id))
-
 		if _, err := c.agents[i].AddUsers(inc); err != nil {
-			// Most probably the agent restarted, so we just start the agent again.
-			if errors.Is(err, client.ErrAgentNotFound) {
-				if _, err := c.agents[i].Run(); err != nil {
-					c.log.Error("agent restart failed", mlog.Err(err))
-				}
-				continue
+			// Most probably the agent crashed, so we just start it again.
```

Comment:
Is it okay not to check for `ErrAgentNotFound` and run the agent always?

Reply:
For our purposes I would say yes. 

I had to move the creation part into the status function cause that's being called by the coordinator before any action is taken. Here we assume that if we get an error it's because somehow the agent crashed and was re-created by the `Status()` method. 

To be extra precise, we would have to fetch the status, check it's not running and only then do the call but the result would be pretty much the same. If it fails we'll get an error and it means something else is wrong.